flashlight/fl/examples/RnnClassification.cpp

```diff
@@ -151,21 +151,49 @@ class RnnClassifier : public Container {
       unsigned vocabSize,
       int hiddenSize = 256,
       unsigned numLayers = 2)
-      : embed_(Embedding(hiddenSize, vocabSize)),
-        rnn_(
-            RNN(hiddenSize,
-                hiddenSize,
-                numLayers,
-                RnnMode::GRU,
-                false /* Dropout */)),
-        linear_(Linear(hiddenSize, numClasses)),
+      : embed_(std::make_shared<Embedding>(hiddenSize, vocabSize)),
+        rnn_(std::make_shared<RNN>(
+            hiddenSize,
+            hiddenSize,
+            numLayers,
+            RnnMode::GRU,
+            0 /* Dropout */)),
+        linear_(std::make_shared<Linear>(hiddenSize, numClasses)),
         logsoftmax_(0) {
     std::cout << "Creating a RNN Classifier with vocab size: " << vocabSize
               << " and num classes: " << numClasses << std::endl;
+    createLayers();
+  }
+
+  RnnClassifier(const RnnClassifier& other) {
+    copy(other);
+    createLayers();
+  }
+
+  RnnClassifier(RnnClassifier&& other) = default;
+
+  RnnClassifier& operator=(const RnnClassifier& other) {
+    clear();
+    copy(other);
+    createLayers();
+  }
+  RnnClassifier& operator=(RnnClassifier&& other) = default;
+
+  void copy(const RnnClassifier& other) {
+    embed_ = std::make_shared<Embedding>(*other.embed_);
+    rnn_ = std::make_shared<RNN>(*other.rnn_);
+    linear_ = std::make_shared<Linear>(*other.linear_);
+    logsoftmax_ = other.logsoftmax_;
+  }
+
+  void createLayers() {
     add(embed_);
     add(rnn_);
     add(linear_);
-    add(logsoftmax_);
+  }
+
+  std::unique_ptr<Module> clone() const override {
+    throw std::make_unique<RnnClassifier>(*this);
```

Comment:
I think this is supposed to be `return`? :)
```suggestion
    return std::make_unique<RnnClassifier>(*this);
```

Reply:
Yeah, find replace got me!